vyper/semantics/analysis/local.py

```diff
@@ -347,66 +347,62 @@ def visit_For(self, node):
                 )
             range_ = node.iter
             validate_call_args(range_, (1, 2), kwargs=["bound"])
+            bound = None
+            if range_.keywords:
+                bound = range_.keywords[0].value  # this is the only kwarg accepted by validation
```

Comment:
i'd prefer to leave the old way (constructing a dict out of kwargs) because it scales better

Reply:
Fixed